### ISSUE TITLE
Error: Getting metadata from plugin failed with error: invalid_grant

### ISSUE THREAD

**Issue by aidensSandbox** [Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client#449]
*Error: Getting metadata from plugin failed with error: invalid_grant*

My internet is working, all my other apps are working, and I have internet.

I'm not sure why it decided to not work anymore.

It was working for 2 weeks before with no issues.

**Describe the bug**

**To Reproduce**
1. Open the unofficial Google Assistant app
2. Type anything, example: "Hi"
3. See error: " You are offline! Please check your internet connection... Error: Getting metadata from plugin failed with error: invalid_grant"

**Screenshots**
https://imgur.com/a/pprZxzW

**Desktop (please complete the following information):**
 - OS: Bigsur 11.2.3

**Comment by sumeshir26** [comment]
Hi @aidensSandbox, please find the fix [in the FAQ's](https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client/wiki/Frequently-Asked-Questions-(FAQ)#how-do-i-fix-invalid_grant-error).

**Comment by coder-lg** [comment]
Simple. Step 1: Find your `tokens.json` file you used in Google Assistant.
Step 2: Move it to the Trash folder and clear that file PERMANENTLY.
Step 3: Open the app, reauthorize and use the app.

**Comment by jamescastells** [comment]
It now stays in this page:

<img width="1017" alt="image" src="https://user-images.githubusercontent.com/5696029/126715980-56941943-c77e-40a7-968a-9f80c96d46d7.png">

This app doesn't work anymore. Uninstalling...

**Comment by coder-lg** [comment]
Check the FAQ in the Wiki in the browser app of GitHub in your Safari. Solutions are there. If you want the latest edition of GAUDC, you have to build it via NPM.

**Comment by coder-lg** [comment]
Latest edition or latest commit edition can be built, a new release is coming soon. Till then, refer the FAQ.

**Comment by coder-lg** [comment]
Turn off `Enable Microphone On Startup`, login and then you can re-enable it

**Comment by chambm** [comment]
Deleting the token file and reauthorizing is a workaround for this issue, but not really a fix. I don't use this app every day. Right now I use it maybe once a week. So when I activate it I'd like it to just work. Having to delete the token file and reauthorize practically every time I use it makes it seem like I'm assisting it more than it's assisting me.. :P But it's a fantastic app when it does work: I love being able to control Google Home devices from my PC. Is there a workaround which will prevent this invalid_grant issue from happening? Or is there a fix in the works?

**Comment by Melvin-Abraham** [comment]
@chambm The issue has been addressed by the commit e635243. This directly implements the option to reset the tokens without having to leave the app and delete it manually from the particular location.

The above commit is not a part of latest release yet but will be released soon. Until then, you may test it out by cloning the repo and building it yourself if you are curious :)

**Comment by chambm** [comment]
Had to install npm to build the installer, but it was worth it.  :joy:

**Comment by im-coder-lg** [comment]
Yeah, since this is a JavaScript program that uses Electron and has a package.json file, it is pretty clear that you need NPM and that too, the one that comes with Node JS.
